astropy/table/_dataframes.py

```diff
@@ -0,0 +1,479 @@
+
+
+def to_df(
+    table: "Table",
+    backend: str | types.ModuleType,
+    index: bool | str | None = None,
+    use_nullable_int: bool = True,
+):
+    """Convert an Astropy Table to a DataFrame using the specified backend."""
+    try:
+        import narwhals as nw
+    except ImportError:
+        raise ImportError(
+            "The narwhals library is required for the generic to_df method. "
+            "If you want to only convert to pandas, use the `to_pandas` method instead."
+        )
+
+    backend_impl = _get_backend_impl(backend)
+
+    if not nw._utils.is_eager_allowed(backend_impl):
+        raise ValueError("Must export to eager compatible DataFrame")
+
+    # Handle index argument
+    index = _handle_index_argument(table, index, backend_impl)
+
+    # Encode mixins and validate columns
+    tbl = _encode_mixins(table)
+    is_masked = tbl.has_masked_columns
+    _validate_columns_for_backend(tbl, backend_impl)
+
+    # Convert to narwhals DataFrame
+    array = tbl.as_array()
+    array_dict = {n: array[n].data if is_masked else array[n] for n in tbl.colnames}
+    df = nw.from_dict(array_dict, backend=backend_impl)
+
+    # Handle masked columns
+    if is_masked:
+        masked_cols = [
+            name
+            for name, col in tbl.columns.items()
+            if isinstance(col, MaskedColumn) and col.mask.any()
+        ]
+        old_dtypes = {n: df[n].dtype for n in masked_cols}
+
+        df = df.with_columns(
+            [
+                (
+                    nw.when(
+                        nw.from_numpy(
+                            array[n].mask.reshape(-1, 1), backend=backend_impl
+                        )[:, 0]
+                    )
+                    .then(None)
+                    .otherwise(nw.col(n))
+                    .alias(n)
+                )
+                for n in masked_cols
+            ]
+        )
+
+        for n in masked_cols:
+            if old_dtypes[n].is_integer() and not use_nullable_int:
+                raise ValueError(
+                    "Cannot convert masked integer columns to DataFrame without using nullable integers. "
+                    f"Set use_nullable_int=True or remove the offending column: {n}."
+                )
+            elif old_dtypes[n].is_float():
+                df = df.with_columns(nw.col(n).cast(old_dtypes[n]).alias(n))
+
+    # Convert to dataframe
+    df = df.to_native()
+
+    # Fix pandas-like nullable integers
+    if backend_impl.is_pandas_like() and is_masked and use_nullable_int:
+        for name in masked_cols:
+            dtype = array[name].dtype
+            if dtype.kind in "iu":
+                new_dtype = dtype.name.replace("i", "I").replace("u", "U")
+                df[name] = df[name].astype(new_dtype)
+
+    # Pandas-like index
+    if index:
+        df.set_index(index, inplace=True, drop=True)
+
+    return df
+
+
+def from_df(
+    df: Any, index: bool = False, units: Mapping[str, "UnitLike"] | None = None
```

Comment:
(using `from narwhals.typing import DataFrameT`)
```suggestion
    df: DataFrameT,
    index: bool = False, units: Mapping[str, "UnitLike"] | None = None,
```